client/src/views/worldmap/WorldMap.jsx

```diff
@@ -1,83 +1,150 @@
 } from "react-simple-maps"
-import { scaleLinear } from "d3-scale"
-import * as d3 from "d3";
+
+// Import Components
+import SliderComponent from '../components/SliderComponent'
+
+// Import Modules
+import Api from '../../modules/api';
 
 const styles = {
   root: {
     display: 'flex',
     margin: 'auto',
     width: 1000,
-    marginTop: '170px'
+    marginTop: '170px',
+    flexDirection: 'column',
   }
 };
 
 class WorldMap extends Component {
-  getRandomColour = (length) => {
-    var clr = d3.scaleLinear()
-    .range(["white", "black"])
-    .domain([0, length]);
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: [],
+      year: 2016,
+      serverUrl: 'https://9fwwprofzl.execute-api.ap-southeast-2.amazonaws.com/dev',
+      chartUrl: "https://raw.githubusercontent.com/zcreativelabs/react-simple-maps/master/examples/choropleth-map/static/world-50m-with-population.json"
+    }
+  }
 
-    let colourArray = d3.range(length).map(function(d) {
-      return clr(d)  
+  async componentWillMount() {
+    const { year, serverUrl } = this.state;
+    const rangeData = await Api(`${serverUrl}/getRange/${year}`, 'GET');
+
+    await this.getEmissions();
+    this.setState({
+      range: rangeData.body[0]
     })
+  }
 
-    return colourArray[Math.floor(Math.random()*colourArray.length)];
+  calculateEmissions = (country) => {
+    const { data } = this.state;
+    const countryData = data.find(entry => entry.entity === country);
+    const emissionCount = countryData && countryData.emissions || 0;
+    return emissionCount;
   }
-  
+
+  getEmissions = async () => {
+    const { year, serverUrl } = this.state;
+    const emissionData = await Api(`${serverUrl}/getEmissions/${year}`, 'GET');
+    this.setState({ data: emissionData.body })
+  }
+
+  renderHeatmap = (country) => {
+    const { range, data } = this.state;
+    const { max } = range;
+
+    const emissionCount = data.length && this.calculateEmissions(country);
+    const maxShorten = emissionCount / max * 100;
+
+    const red = 'rgb(255, 0, 0)';
+    const yellow = 'rgb(255, 255, 0)';
+    const green = 'rgb(0, 255, 0)';
+    const grey = 'rgb(66, 66, 66)';
+
+    if (maxShorten < 15) {
+      return green
+    } else if (maxShorten < 30) {
+      return yellow;
+    } else if (maxShorten > 30) {
+      return red;
+    } else {
+      return grey;
+    }
+  }
+
+  updateYear = async (event, value) => {
+    this.setState({ year: value })
+    await this.getEmissions();
+  }
+
   render() {
-      const { classes } = this.props;
-      return (
-        <div className={classes.root}>
-          <ComposableMap
-            projectionConfig={{
-              scale: 205,
-              rotation: [-11,0,0],
-            }}
-            width={980}
-            height={551}
-            style={{
-              width: "100%",
-              height: "auto",
-            }}
+    const { classes } = this.props;
+    const { data, year, map_url } = this.state;
```

Comment:
Where is this defined? I think you removed it when you merged changes
Also, we have a standard of camelCase in this project instead of snake_case
Could you please stick to it 👍

Reply:
Solved. Yep the wrong variable was being used, hence the map not loading.